### ISSUE TITLE
HTTP Upload fails with invalid-status-line 

### ISSUE THREAD

**Issue by bill-mcgonigle** [siacs/Conversations#2279]
#### General information

* **Version:** eg 1.16.0
* **Device:** Moto X Pure
* **Android Version:** Android 6.0 Stock
* **Server name:** self-hosted
* **Server software:**  prosody 0.10 nightly 299
* **Installed server modules:**  Stream Managment, message carbons, MAM
* **Conversations source:** PlayStore Beta

#### Steps to reproduce

1.try to send picture with 1.16.0.
2.fall back to 1.15.2 and it works fine

#### Expected result

sent picture

Feb 06 23:44:59 socket  debug   server.lua: ssl handshake done
Feb 06 23:44:59 http.server     debug   error_cb: invalid-status-line
Feb 06 23:44:59 socket  debug   server.lua: closed client handler and removed socket from list

#### Debug output

````
nothing showing in logcat from that timeframe
````

**Comment by github-k8n** [comment]
Same issue here, seems like there is a mismatch between what prosody thinks the slot/path should be and what conversations is using.. maybe due to file extension?
Better detailed log from prosody (replaced hostname and JID):
```
2017-02-10T10:33:08+01:00 router prosody[6454]: c2s733a3295f0: Sending <r> (after send)
2017-02-10T10:33:08+01:00 router prosody[6454]: c2s733a3295f0: Handled 39 incoming stanzas
2017-02-10T10:33:08+01:00 router prosody[6454]: c2s733a3295f0: Received[c2s]: <iq id='7m9uk2unjd' type='get' to='HOSTNAME' from='JID'>
2017-02-10T10:33:08+01:00 router prosody[6454]: c2s733a3295f0: #queue = 2
2017-02-10T10:33:08+01:00 router prosody[6454]: c2s733a3295f0: Given upload slot "4ef7dbf0-7e8e-4f9b-a2df-fd72089ab093"
2017-02-10T10:33:08+01:00 router prosody[6454]: c2s733a3295f0: Received[c2s]: <a xmlns='urn:xmpp:sm:3' h='372'>
2017-02-10T10:33:08+01:00 router prosody[6454]: c2s733a3295f0: #queue = 1
2017-02-10T10:33:08+01:00 router prosody[6454]: socket: server.lua: accepted new client connection from ::1:42280 to 5280
2017-02-10T10:33:09+01:00 router prosody[6454]: http.server: Firing event: PUT HOSTNAME/upload/4ef7dbf0-7e8e-4f9b-a2df-fd72089ab093/RInaC-ZbT-KSpbK_N0W2lg==
2017-02-10T10:33:09+01:00 router prosody[6454]: HOSTNAME:http_upload: Attempt to upload to unknown slot "4ef7dbf0-7e8e-4f9b-a2df-fd72089ab093/RInaC-ZbT-KSpbK_N0W2lg=="
2017-02-10T10:33:09+01:00 router prosody[6454]: socket: server.lua: closed client handler and removed socket from list
```

**Comment by iNPUTmice** [comment]
should be fixed in 54ae8abcb72b777d7886bf3aa6f3b1abd24788f8
